scripts/o3de/o3de/repo_properties.py

```diff
@@ -293,6 +293,44 @@ def _auto_update_json(object_type: str or list,
             _edit_objects(object_name, validation.valid_o3de_template_json, repo_json, expected_files.get("template.json"))
     return 0
 
+def print_repo_diff(repo_json: dict,
+             repo_json_original: dict):
+    
+    if repo_json_original == repo_json:
+            logger.warning('No difference is found between your current repo.json file and your dry run')
+            return 0
+        
+    pretty_print_string = []
+    pretty_print_string.append('Dry Run:')
+    for key in repo_json:
+        # get the field that changed
+        if repo_json_original[key] != repo_json[key]:
+            object_key = key.replace('s_data', '')
+            object_original = {object[f'{object_key}_name'] for object in repo_json_original.get(key, [])}
+            object_dry_run = {object[f'{object_key}_name'] for object in repo_json.get(key, [])}
+            # check for new objects
+            objects_added = object_dry_run - object_original
+            pretty_print_string.append(f"{object_key.capitalize()}s Added: {len(objects_added)}")
+            for i, object_name in enumerate(objects_added, start=1):
+                pretty_print_string.append(f"  {i}. {object_name}")
+
+            # when object name is the same check if there exist a version_data field in the dry-run update
+            modified_objects = [
+                dry_run_repo[f'{object_key}_name']
+                for dry_run_repo in repo_json.get(key, [])
+                for original_repo in repo_json_original.get(key, [])
+                if dry_run_repo[f'{object_key}_name'] == original_repo[f'{object_key}_name']
+                and any(field not in original_repo for field in dry_run_repo)
+            ]    
+            if modified_objects:
+                pretty_print_string.append(f'{object_key.capitalize()}s Modified: {len(modified_objects)}')
+                for i, object_name in enumerate(modified_objects, start=1):
+                    pretty_print_string.append(f"  {i}. {object_name}")
+            else:
+                pretty_print_string.append(f'{object_key.capitalize()}s Modified: {len(modified_objects)}')
+    logger.warning('\n'.join(pretty_print_string))
```

Comment:
This isn't an appropriate use for a `logger.warning`. The function's purpose is to print, so we should use `print` here instead of `logger.warning`